kbase-extension/static/kbase/js/common/jobs.js

```diff
@@ -332,43 +343,63 @@ define(['common/errorDisplay', 'common/format', 'common/html', 'common/props', '
     }
 
     /**
-     * Given an array of jobState objects, create an object with jobs indexed by ID and by status
+     * Add an array of jobs to a model
+     *
+     * @param {array} jobArray - array of job state objects
+     * @param {object} model - the model to add the jobs to
+     */
+    function populateModelFromJobArray(jobArray = [], model) {
+        if (!model) {
+            throw new Error('Missing a model to populate');
+        }
+        const batchIds = new Set(),
+            batchJobs = new Set();
+
+        if (!jobArray.length) {
+            model.deleteItem('exec.jobs');
+            model.deleteItem('exec.jobState');
+            return;
+        }
+
+        jobArray.forEach((job) => {
+            if (job.batch_id) {
+                batchIds.add(job.batch_id);
+            }
+            if (job.batch_job) {
+                batchJobs.add(job);
+            }
+        });
+
+        if (batchIds.size > 1) {
+            // more than one batch present
+            throw new Error('More than one batch ID found');
+        }
+        if (batchJobs.size > 1) {
+            // more than one batch parent present
+            throw new Error('More than one batch parent found');
+        }
+
+        model.setItem('exec.jobs', jobArrayToIndexedObject(jobArray));
+
+        model.setItem('exec.jobState', Array.from(batchJobs)[0]);
+        return model;
+    }
+
+    /**
+     * Given an array of jobState objects, create an object with jobs indexed by ID
      *
      * @param {array} jobArray array of jobState objects
      * @returns {object} with indexed job data:
      *      byId        key: job_id, value: jobState object
-     *      byStatus    key: status, value: object whose keys are the job IDs
-     *                      of jobs with that status
      */
 
     function jobArrayToIndexedObject(jobArray = []) {
         const jobIx = {
             byId: {},
-            byStatus: {},
-            jobsWithRetries: {},
-            batchId: null,
         };
 
         jobArray.forEach((job) => {
-            const jobId = job.job_id,
-                status = job.status;
-            jobIx.byId[jobId] = job;
-            if (!jobIx.byStatus[status]) {
-                jobIx.byStatus[status] = {};
-            }
-            jobIx.byStatus[status][jobId] = true;
-
-            // any job with one or more retries
-            if (job.retry_ids && job.retry_ids.length) {
-                jobIx.jobsWithRetries[jobId] = true;
-            }
-
-            if (job.batch_job) {
-                if (jobIx.batchId && jobId !== jobIx.batchId) {
-                    console.error(`Error: two batch IDs present: ${jobIx.batchId} and ${jobId}`)
-                }
-                jobIx.batchId = job.batch_id;
-            }
+            jobIx.byId[job.job_id] = job;
```

Comment:
just index by ID for now. I hope that I didn't have some specific purpose in mind for those other indexes!